### PR TITLE
Ftr: add TLS support

### PR DESCRIPTION
**Pull request by aliiohs**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module github.com/apache/dubbo-go
 
+go 1.14
```
Comment (AlexStocks) [inline go.mod:3]: pld do not add go version in the go.mod.

File: go.mod [hunk 2]
```diff
@@ -66,4 +68,4 @@ require (
 	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a // indirect
 )
 
-go 1.13
+replace github.com/dubbogo/getty v1.2.2 => github.com/aliiohs/getty v1.1.1-0.20200802094147-169328c4ff38
```
Comment (zouyx) [inline go.mod:71]: remove this replace ?

**Comment by zouyx** [comment]
Fix travis failing

File: go.mod [hunk 3]
```diff
@@ -58,12 +58,11 @@ require (
...
-	gopkg.in/mgo.v2 v2.0.0-20180705113604-9856a29383ce // indirect
 	gopkg.in/yaml.v2 v2.2.8
 	k8s.io/api v0.16.9
 	k8s.io/apimachinery v0.16.9
 	k8s.io/client-go v0.16.9
 	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a // indirect
 )
 
-go 1.13
+replace github.com/dubbogo/getty v1.3.7 => github.com/aliiohs/getty v1.1.1-0.20200802094147-169328c4ff38
```
Comment (zouyx) [inline go.mod:68]: getty has released `1.3.8`

File: go.mod (hunk 1 above)
Comment (zouyx) [reply go.mod:3]: agree

File: protocol/dubbo/config.go [hunk 4]
```diff
@@ -18,11 +18,11 @@
 package dubbo
 
 import (
+	"github.com/apache/dubbo-go/config"
```
Comment (zouyx) [inline protocol/dubbo/config.go:21]: split pkg

File: protocol/dubbo/config.go [hunk 5]
```diff
@@ -22,9 +22,11 @@ import (
 )
 
 import (
-	"github.com/dubbogo/getty"
 	perrors "github.com/pkg/errors"
 )
+import (
```
Comment (zouyx) [inline protocol/dubbo/config.go:27]: add a blank line

File: config/config_loader.go [hunk 6]
```diff
@@ -325,6 +326,12 @@ func GetBaseConfig() *BaseConfig {
 	return baseConfig
 }
 
+func GetSslEnabled() bool {
+	return sslEnabled
+}
+func SetSslEnabled(enabled bool) {
```
Comment (zouyx) [inline config/config_loader.go:332]: Is it necessary for change this flag？

File: protocol/dubbo/pool.go [hunk 7]
```diff
@@ -53,15 +55,27 @@ var (
 )
 
 func newGettyRPCClientConn(pool *gettyRPCClientPool, protocol, addr string) (*gettyRPCClient, error) {
-	c := &gettyRPCClient{
-		protocol: protocol,
-		addr:     addr,
-		pool:     pool,
-		gettyClient: getty.NewTCPClient(
+	var gettyClient getty.Client
+	if pool.sslEnabled {
```
Comment (zouyx) [inline protocol/dubbo/pool.go:59]:
```suggestion
	var (
	      gettyClient getty.Client
	      sslEnabled bool
        )
	sslEnabled = pool.sslEnabled
	if sslEnabled {
```

make a comment variable.

File: protocol/dubbo/pool.go [hunk 8]
```diff
@@ -103,7 +117,23 @@ func (c *gettyRPCClient) newSession(session getty.Session) error {
 	if conf.GettySessionParam.CompressEncoding {
 		session.SetCompressType(getty.CompressZip)
 	}
-
+	if c.pool.sslEnabled {
```
Comment (zouyx) [inline protocol/dubbo/pool.go:120]:
```suggestion
	if sslEnabled {
```

File: config/consumer_config.go [hunk 9]
```diff
@@ -34,6 +33,10 @@ import (
 	"github.com/apache/dubbo-go/common/yaml"
 )
 
+const (
+	MaxWheelTimeSpan = 900e9 // 900s, 15 minute
```
Comment (AlexStocks) [inline config/consumer_config.go:37]: if u need this param, why submit a pr to getty to expose it for dubbogo?

Comment (zouyx) [reply config/consumer_config.go:37]: honestly , I don't think this is a good idea.

In this case, getty developer changed the variable scope, we don't know.

So if this variable scope is `dubbo-go` , i suggest keep it in `dubbo-go` is better.

File: config/config_loader.go (hunk 6 above)
Comment (aliiohs) [reply config/config_loader.go:332]: yes,this flag will add to url，and use in client

File: protocol/dubbo/server.go [hunk 10]
```diff
@@ -126,7 +128,20 @@ func (s *Server) newSession(session getty.Session) error {
...
+		session.SetName(conf.GettySessionParam.SessionName)
+		session.SetMaxMsgLen(conf.GettySessionParam.MaxMsgLen)
+		session.SetPkgHandler(rpcServerPkgHandler)
+		session.SetEventListener(s.rpcHandler)
+		session.SetWQLen(conf.GettySessionParam.PkgWQSize)
+		session.SetReadTimeout(conf.GettySessionParam.tcpReadTimeout)
+		session.SetWriteTimeout(conf.GettySessionParam.tcpWriteTimeout)
+		session.SetCronPeriod((int)(conf.sessionTimeout.Nanoseconds() / 1e6))
+		session.SetWaitTime(conf.GettySessionParam.waitTimeout)
+		logger.Debugf("app accepts new session:%s\n", session.Stat())
```
Comment (gaoxinge) [inline protocol/dubbo/server.go:141]: It's better to change `app` to `server`.

File: config/consumer_config.go [hunk 11]
```diff
@@ -19,12 +19,12 @@ package config
 
 import (
 	"bytes"
+	"github.com/dubbogo/getty"
 	"time"
 )
 
 import (
 	"github.com/creasty/defaults"
-	"github.com/dubbogo/getty"
```
Comment (gaoxinge) [inline config/consumer_config.go:27]: Place getty import here.

**Pull request merged**